components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -966,36 +965,36 @@ public AccessTokenDO getAccessToken(String accessTokenIdentifier, boolean includ
         PreparedStatement prepStmt = null;
         ResultSet resultSet = null;
 
+        if (StringUtils.isEmpty(tenantDomain)) {
+            throw new IdentityOAuth2Exception("Tenant Domain should be needed for further execution");
+        }
+
         try {
             String sql;
 
             if (includeExpired) {
                 if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
-                    if ((isTenantQualifiedUrlsEnabled && !isCrossTenantTokenInspectionAllowed)
-                            && tenantDomain != null) {
+                    if (!isCrossTenantTokenInspectionAllowed) {
                         sql = SQLQueries.RETRIEVE_ACTIVE_EXPIRED_TENANT_ACCESS_TOKEN_IDP_NAME;
                     } else {
                         sql = SQLQueries.RETRIEVE_ACTIVE_EXPIRED_ACCESS_TOKEN_IDP_NAME;
                     }
                 } else {
-                    if ((isTenantQualifiedUrlsEnabled && !isCrossTenantTokenInspectionAllowed)
-                            && tenantDomain != null) {
+                    if (!isCrossTenantTokenInspectionAllowed) {
```

Comment:
this check needs to be moved to the service layer

Reply:
I have created a separate issue for this https://github.com/wso2/product-is/issues/14269

Reply:
And Created an issue for change the default value of the config https://github.com/wso2/product-is/issues/14270